src/main/java/com/softserve/academy/event/service/db/impl/StatisticServiceImpl.java

```diff
@@ -65,20 +67,20 @@ public QuestionsGeneralStatisticDTO getGeneralStatistic(Long id) {
     @Override
     @Transactional
     public void isSurveyBelongsUser(Long surveyId) {
-        Object principal = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
-        if (!(principal instanceof UserDetails)) {
-            log.info("Throw UnauthorizedException");
-            throw new UnauthorizedException();
-        }
-        UserDetails userDetails = (UserDetails) principal;
+//        Object principal = SecurityContextHolder.getContext().getAuthentication().getPrincipal(); // unneeded, read SecurityUserUtils!
+//        if (!(principal instanceof UserDetails)) {
+//            log.info("Throw UnauthorizedException");
+//            throw new UnauthorizedException();
+//        }
+//        UserDetails userDetails = (UserDetails) principal;
```

Comment:
```suggestion
//        UserDetails userDetails = (UserDetails) principal;
```
What?

Reply:
If user authorize using oAuth2 this peace of code drop UnauthorizedException, but user authorized!

Reply:
Working user details you can get from SecurityUserUtil.getCurrentUserEmail